### PR TITLE
Override type reader

### PR DESCRIPTION
**Pull request by Joe4evr**
This is the biggest contribution I've made yet, and I don't know if I got all the cases. Please do mention if I missed anything.

### CODE REVIEW

File: src/Discord.Net.Commands/Module.cs [hunk 1]
```diff
@@ -16,6 +19,7 @@ public class Module
         public string Remarks { get; }
         public IEnumerable<Command> Commands { get; }
         internal object Instance { get; }
+        public ConcurrentDictionary<Type, TypeReader> OverridingTypeReaders { get; }
```
Comment (RogueException) [inline src/Discord.Net.Commands/Module.cs]: No need for this to be concurrent, it should be immutable after the module has been created

Comment (Joe4evr) [inline src/Discord.Net.Commands/Module.cs]: Roger. Will change that in the morning.

**Comment by RogueException** [comment]
The rest looks LGTM, has this been tested?

**Comment by Joe4evr** [comment]
*Can* I even test?

**Comment by khionu** [comment]
How could you *not*? O.o

**Comment by Joe4evr** [comment]
Pinging @RogueException, I updated it to fit the new stuff.

**Comment by Joe4evr** [comment]
Closing, as @FiniteReality is taking over.

**Pull request closed without merging**